libcloud/compute/drivers/openstack.py

```diff
@@ -1551,7 +1561,9 @@ def create_node(
         if ex_os_scheduler_hints:
             data["os:scheduler_hints"] = ex_os_scheduler_hints
 
-        resp = self.connection.request("/servers", method="POST", data=data)
+        resp = self.connection.request(
+            "/servers", method="POST", data=data, headers=self._set_microversion()
```

Comment:
Hm, so we only want to set this header for "POST /servers" requests and nothing else? Shouldn't it be set for all the outgoing API requests?

Reply:
And in case we do want to set it for all the outgoing requests, ``add_default_headers()`` on the connection class may be a better place to do that.